### PR TITLE
Support building dependencies with Emscripten

### PR DESCRIPTION
**Pull request by rhuffy**
These dependencies can be compiled to WASM with modified build flags. Specifically, this fixes builds for:
- dsfmt
- gmp
- libuv
- mpfr
- pcre

### CODE REVIEW

**Review by vchuravy — APPROVED** [review]

File: deps/pcre.mk [hunk 1]
```diff
@@ -6,6 +6,13 @@ ifneq ($(USE_BINARYBUILDER_PCRE),1)
 PCRE_CFLAGS := -O3
 PCRE_LDFLAGS := $(RPATH_ESCAPED_ORIGIN)
 
+ifeq ($(OS),emscripten)
+PCRE_CFLAGS += -fPIC
+PCRE_JIT = --disable-jit
```
Comment (staticfloat) [inline deps/pcre.mk:11]: I was under the impression that the PCRE JIT is not optional for us.  Is that no longer true?

File: deps/gmp.mk [hunk 2]
```diff
@@ -19,6 +19,9 @@ ifeq ($(BUILD_OS),WINNT)
 GMP_CONFIGURE_OPTS += --srcdir="$(subst \,/,$(call mingw_to_dos,$(SRCCACHE)/gmp-$(GMP_VER)))"
 endif
 
+ifeq ($(OS),emscripten)
+GMP_CONFIGURE_OPTS += CFLAGS="-fPIC"
```
Comment (staticfloat) [inline deps/gmp.mk:23]: It might be worthwhile adding a comment stating why it's important to have PIC for GMP on emscripten

**Review by staticfloat — APPROVED** [review]
While I have some small reservations about including more and more build options into our main tree, I don't have a good alternative (ideally we'd have some way of building these projects externally, then just dropping them into the build tree).  So just a few small comments, and I think this is okay to merge.

File: deps/pcre.mk (hunk 1 above)
Comment (vtjnash) [reply deps/pcre.mk:11]: It is optional, just a bit slower without it

**Pull request merged**